libimage/runtime.go

```diff
@@ -22,13 +23,16 @@ import (
 var json = jsoniter.ConfigCompatibleWithStandardLibrary
 
 // tmpdir returns a path to a temporary directory.
-func tmpdir() string {
-	tmpdir := os.Getenv("TMPDIR")
-	if tmpdir == "" {
-		tmpdir = "/var/tmp"
+func tmpdir() (string, error) {
+	var tmpdir string
+	defaultContainerConfig, err := config.Default()
+	if err == nil {
```

Comment:
Non-blocking style nit:  I prefer early returns `if err != nil { return ... }`

Reply:
Me too, this was because of the earlier design were it was just logrus.Warning.